### PR TITLE
WIP: Add identicon into profile qr code

### PR DESCRIPTION
**Pull request by Ferossgp**
Added only to profile for now

### CODE REVIEW

**Comment by Samyoul** [comment]
Would you mind sharing a sample of the QRs that are produced with this change? Thank you 😄

**Review by hesterbruikman — COMMENTED** [review]
Looks great on my end! Tried on Android. @errorists can you also take a look please?

**Comment by Ferossgp** [comment]
image

File: src/status_im/ui/components/qr_code_viewer/views.cljs [hunk 1]
```diff
@@ -6,23 +6,63 @@
...
+        logo-position       (/ (- size logo-size) 2)
+        background-position (/ (- size background-size) 2)]
+    [svg/g
+     [svg/defs
+      [svg/clip-path {:id "logo-clip"}
+       [svg/rect {:width         logo-size
+                  :height        logo-size
+                  :stroke        "blue"
+                  "stroke-width" 2
+                  :rx            (/ logo-size 2)
```
Comment (rasom) [inline src/status_im/ui/components/qr_code_viewer/views.cljs:21]: can we move `(/ logo-size 2)` to `let`? It is repeated four times

File: src/status_im/ui/components/qr_code_viewer/views.cljs [hunk 2]
```diff
@@ -6,23 +6,63 @@
...
+                  :height              logo-size
+                  :href                identicon
+                  :clip-path           "url(#logo-clip)"
+                  :preserveAspectRatio "xMidYMid slice"}]]]))
+
+(defn qr-code [{:keys [size value identicon]}]
+  (let [path (reagent/atom nil)]
+    (.toString qr-code-js
+               value
+               (bean/->js {:margin 0 :width size :errorCorrectionLevel "H"})
```
Comment (rasom) [inline src/status_im/ui/components/qr_code_viewer/views.cljs:45]: Can we either have only one key/value pair per line or separate pairs with comma? It is a bit hard to read for me personally

Comment (Ferossgp) [reply src/status_im/ui/components/qr_code_viewer/views.cljs:45]: I totally agree with you, and I would prefer to have that as a lint rule. We have many places where it's multiple pairs per line, and IMHO it's even harder to read when it's multiple pairs per line and multiple lines for the same map.

**Comment by oskarth** [comment]
Small request - could you please rebase to latest master so this commit https://github.com/status-im/status-react/commit/b5fda1234e5447cbc626059dfc78fcfbabc9e926 gets included? This means we aren't running E2E tests on eth.prod cluster anymore, and since this happens on every push, the impact is quite big. This ensures we have more accurate metrics going forward from Jul 1 onward, which would be awesome. See https://discuss.status.im/t/user-growth-and-retention/1782 for more

**Comment by hesterbruikman** [comment]
For reference: This PR is part of epic #10502 and fixes #10627 

PR is currently WIP, using compressed key not implemented yet, method available in status-go see https://specs.status.im/spec/2#public-key-serialization and https://github.com/status-im/status-go/pull/1990. Issue to implement use of compressed key in status-react here: #10325

**Comment by Ferossgp** [comment]
Should be done in Go as it should be shared between all platforms

**Pull request closed without merging**